lib/pure/parseutils.nim

```diff
@@ -287,12 +287,13 @@ proc parseUInt*(s: string, number: var uint, start = 0): int {.
   ## into `number`.
   ## Result is the number of processed chars or 0 if there is no integer or
   ## overflow detected.
-  var res: uint64
+  var res: BiggestUInt
   result = parseBiggestUInt(s, res, start)
-  if (sizeof(uint) <= 4) and
-      (res > 0xFFFF_FFFF'u64):
-    raise newException(OverflowError, "overflow")
-  elif result != 0:
+  when not defined(JS):
```

Comment:
This should be ``when sizeof(BiggestUInt) > sizeof(uint)``, maybe JS gets 64bit unsigned support one day.